src/Nethermind/Nethermind.TxPool/LightTransaction.cs

```diff
@@ -0,0 +1,31 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+
+using Nethermind.Core;
+
+namespace Nethermind.TxPool;
+
+/// <summary>
+/// For sorting reasons - without storing full, large txs in memory
+/// </summary>
+public class LightTransaction : Transaction
+{
+    public LightTransaction(Transaction fullTx)
+    {
+        Type = TxType.Blob;
+        Hash = fullTx.Hash;
+        SenderAddress = fullTx.SenderAddress;
+        Nonce = fullTx.Nonce;
+        Value = fullTx.Value;
+        GasLimit = fullTx.GasLimit;
+        GasPrice = fullTx.GasPrice; // means MaxPriorityFeePerGas
+        DecodedMaxFeePerGas = fullTx.DecodedMaxFeePerGas;
+        MaxFeePerBlobGas = fullTx.MaxFeePerBlobGas;
+        BlobVersionedHashes = new byte[fullTx.BlobVersionedHashes!.Length][];
```

Comment:
Do we need to allocate it? If it is an empty array? It also should have length of 1 or 2? So We should be able to use some constants (statics)?

Reply:
It's needed only for `BlobVersionedHashes.Length` checks, which need to be the same as in full blob tx. It can be 1, 2, 3, 4, 5 or 6. It's not a big allocation, potentially we can create static arrays of lengths 1, 2, 3, 4, 5 and 6 and write some method to pick proper one for considered tx. Not sure if slightly reduced allocations are justifying complicating code in this case